pagefaultgames/pokerogue
# [Improvement] Refactored Test Phase Management with Error Handling

## What are the changes?
Refactored phase management in tests to ensure it resolves when the phase's `end()` method is called. Simplified multiple layout complexities and added error handlers to catch errors and halt async calls.

## Why am I making these changes?
These changes are made to enhance the robustness and reliability of phase management, streamline the application layout, and reduce potential error sources. This refactoring and error handling improvement address issue #123, aiming to improve user experience and system reliability.

## What has changed?
- **Before:** Phase management involved multiple layout complexities and lacked robust error handling.
- **After:** Phase management now correctly resolves with the `end()` method, has a simplified layout, and includes error handlers to catch errors and stop async calls.

## Screenshots
Test that were passing before are still passing now:
![image](https://github.com/pagefaultgames/pokerogue/assets/44787002/dbb25eae-31f0-4343-a780-e82cb720673f)


## How to test the changes?
1. Checkout the branch with these changes.
2. Run the application and navigate to the test section.
3. Verify that the phase correctly resolves with its `end()` method and that errors are properly handled.

## Checklist
- [x] Is there no overlap with another PR?
- [x] Is the PR self-contained and not splittable into smaller PRs?
- [x] Have I clearly explained the changes?
- [x] Have I manually tested the changes?
- [ ] Are there visual changes?
